### DISCUSSION TITLE
Alpha dashboard “Restore Session” not working

### DISCUSSION THREAD

**Discussion by ddbrierton** [LazyVim/LazyVim#753] [Q&A] [answered]
On startup, pressing `l`[**edit** I meant `s` not `l`]  to restore the previous session no longer does anything. There is no error message. However, `<space> q l` functions just fine. So there doesn’t seem to be a problem with the session management provided by persistence.

I had customised my dashboard (to add project support), but I’ve commented that out and that didn’t fix anything.

If I run `:checkhealth` I see ‘which-keys` warning of a conflicting keymap:

```
WhichKey: checking conflicting keymaps ~
- WARNING conflicting keymap exists for mode **"n"**, lhs: **"gc"**
- rhs: ` `
```

Those keys aren’t being set by me.

This is a _fairly_ vanilla install, I haven’t done much configuration other than installing a theme and some of the “extras” included with the project, and I followed the docs for configuring them very carefully.

Any suggestions? I want to get better at learning how to track these problems down myself.

**Comment by folke** [comment]
Restoring your last session is with `s`. `l` is for lazy

**Reply by ddbrierton** [reply]
Apologies, that was just a typo on my part. It is the `s` key which isn’t working.

**Reply by folke** [reply]
Well, it's working perfectly fine for me. I also didn't change anything in lazy of the session plugin, so must be something you changed on your end...

**Reply by ddbrierton** [reply]
Yes, I wasn’t suggesting it was a bug, it is something I’ve done — it was working previously. I was just hoping for some guidance in tracking down what exactly, but if that isn’t an appropriate thing to ask here please let me know and I’ll be sure not to do it again.

**Reply by folke** [reply]
when you're on the dashboard, what does `:verbose nmap s` show?

**Reply by ddbrierton** [reply]
`verbose map s` shows this:

```
n  s           *@:lua require("persistence").load() <CR>
	Last set from Luax  s           * <Lua 133: ~/.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/keys.lua:67>
                 Leap forward to
	Last set from Luao  s           * <Lua 132: ~/.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/keys.lua:67>
                 Leap forward to
	Last set from Luan  s           * <Lua 131: ~/.local/share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/keys.lua:67>
                 Leap forward to
	Last set from Lua
```

**Reply by folke** [reply, marked as answer]
That looks normal. Might be that your session is corrupt.

Did you try going to another directory, opening a file, closing again and see if loading a session then works?

**Reply by ddbrierton** [reply]
Ah. Yes, changing directories and opening some files, quitting and restarting then `s` worked.

Going back to the previous directory, starting nvim and then `s` again did not work. Typing `<space> q l` brought up the files I had opened in the previous directory.

**Reply by folke** [reply]
Yes, that's normal. Ql is open last. Qs is session from current directory.

**Reply by ddbrierton** [reply]
So thank you, it seems to be fine in any other directory than that one I had been in. Is the simplest solution to just delete the session file for that directory in `~/.local/state/nvim/sessions/`?

**Reply by folke** [reply]
That, or just open neovim in the directory, open any file close and reopen.

**Reply by ddbrierton** [reply]
The latter was what I was doing anyway, before I decided to post here. No matter what I changed the session for that directory wasn’t working. I was restarting quite a bit because I was editing my nvim config. It was my home directory if that makes any difference.

**Reply by folke** [reply]
Hmm, that's weird indeed. Then probably deleting the session file will fix it.
